src/_utils/offline/service-worker/sw.template.js

```diff
@@ -0,0 +1,57 @@
+const cacheName = "<%= cacheName %>"
+
+/* eslint-disable quotes */
+const cacheList = JSON.parse('<%= files %>')
+/* eslint-enable quotes */
+
+self.addEventListener("install", (event) => {
+  console.log("[SW]:", "installing")
+  event.waitUntil(
+    caches
+      .open(cacheName)
+      .then((cache) => (
+        cache
+          .addAll(cacheList)
+          // Go from install to active without waiting step
```

Comment:
A service worker lifecycle: 

![image](https://cloud.githubusercontent.com/assets/3049054/14545569/06e3c000-02ca-11e6-9dae-11d425f120d5.png)

Waiting step is when there is already a SW in the page and a new SW was registered. The new one have to go to the waiting step until reload.